CMakeLists.txt

```diff
@@ -7,6 +7,12 @@ if(NOT CMAKE_MODULE_PATH)
 endif()
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+find_package(OpenCL)
+find_package(Qt5Widgets)
+find_package(CUDA)
```

Comment:
This change we could revert for now as we cannot compile CUDA

Reply:
I agree